src/dat.rs

```diff
@@ -0,0 +1,231 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use dusk_bytes::DeserializableSlice;
+
+use std::fs;
+use std::io::Read;
+
+use crate::crypto::decrypt;
+use crate::store;
+use crate::Error;
+use crate::WalletPath;
+
+/// Binary prefix for old Dusk wallet files
+pub const OLD_MAGIC: u32 = 0x1d0c15;
+/// Binary prefix for new binary file format
+pub const MAGIC: u32 = 0x72736b;
```

Comment:
The magic for rusk wallet binary format should be  `0x0072_736b8_0200`

where:
0x0072_736b8 => Rusk magic number
0x0200 => File Type (https://github.com/dusk-network/rusk/wiki/Binary-File-Format#file-types)

If you want to check those fields separately, it's ok but please change MAGIC to `0x0072_736b8`